tests/test_sdist.py::test_contents_unstaged fails outside git clone

The following test fails when run from GitHub archive rather than full `git clone`. IIUC `in_git_repo_context` is supposed to make it use a temporary git repo, so I suppose something goes wrong:

```pytb
_______________________________________________________ test_contents_unstaged ________________________________________________________

package_pure = PosixPath('/tmp/meson-python/tests/packages/pure')
tmpdir = local('/tmp/pytest-of-mgorny/pytest-1/test_contents_unstaged0')

    def test_contents_unstaged(package_pure, tmpdir):
        new_data = textwrap.dedent('''
        def bar():
            return 'foo'
        ''').strip()
    
        with open('pure.py', 'r') as f:
            old_data = f.read()
    
        try:
            with in_git_repo_context(), open('pure.py', 'w') as f, open('crap', 'x'):
                f.write(new_data)
    
>           sdist_path = mesonpy.build_sdist(os.fspath(tmpdir))

/tmp/meson-python/tests/test_sdist.py:52: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/tmp/meson-python/mesonpy/__init__.py:869: in build_sdist
    return project.sdist(out).name
/tmp/meson-python/mesonpy/__init__.py:794: in sdist
    self._meson('dist', '--allow-dirty', '--no-tests', '--formats', 'gztar')
/tmp/meson-python/mesonpy/__init__.py:468: in _meson
    return self._proc('meson', *args)
/tmp/meson-python/mesonpy/__init__.py:463: in _proc
    subprocess.check_call(list(args))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

popenargs = (['meson', 'dist', '--allow-dirty', '--no-tests', '--formats', 'gztar'],), kwargs = {}, retcode = 1
cmd = ['meson', 'dist', '--allow-dirty', '--no-tests', '--formats', 'gztar']

    def check_call(*popenargs, **kwargs):
        """Run command with arguments.  Wait for command to complete.  If
        the exit code was zero then return, otherwise raise
        CalledProcessError.  The CalledProcessError object will have the
        return code in the returncode attribute.
    
        The arguments are the same as for the call function.  Example:
    
        check_call(["ls", "-l"])
        """
        retcode = call(*popenargs, **kwargs)
        if retcode:
            cmd = kwargs.get("args")
            if cmd is None:
                cmd = popenargs[0]
>           raise CalledProcessError(retcode, cmd)
E           subprocess.CalledProcessError: Command '['meson', 'dist', '--allow-dirty', '--no-tests', '--formats', 'gztar']' returned non-zero exit status 1.

/usr/lib/python3.11/subprocess.py:402: CalledProcessError
-------------------------------------------------------- Captured stdout call ---------------------------------------------------------
! Using Meson to generate the project metadata (no `project` section in pyproject.toml)
+ meson setup --native-file=/tmp/meson-python/tests/packages/pure/.mesonpy-native-file.ini -Ddebug=false -Dstrip=true -Doptimization=2 --prefix=/usr /tmp/meson-python/tests/packages/pure /tmp/meson-python/tests/packages/pure/.mesonpy-qovklooe/build
The Meson build system
Version: 0.62.2
Source dir: /tmp/meson-python/tests/packages/pure
Build dir: /tmp/meson-python/tests/packages/pure/.mesonpy-qovklooe/build
Build type: native build
Project name: pure
Project version: 1.0.0
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program python3 found: YES (/tmp/meson-python/.venv/bin/python)
Build targets in project: 0

pure 1.0.0

  User defined options
    Native files: /tmp/meson-python/tests/packages/pure/.mesonpy-native-file.ini
    debug       : false
    optimization: 2
    prefix      : /usr
    strip       : true

Found ninja-1.10.2.git.kitware.jobserver-1 at /tmp/meson-python/.venv/bin/ninja
+ meson dist --allow-dirty --no-tests --formats gztar
Dist currently only works with Git or Mercurial repos

```

--- a/tests/test_sdist.py
+++ b/tests/test_sdist.py
@@ -46,10 +46,11 @@ def test_contents_unstaged(package_pure, tmpdir):
         old_data = f.read()
 
     try:
-        with in_git_repo_context(), open('pure.py', 'w') as f, open('crap', 'x'):
-            f.write(new_data)
+        with in_git_repo_context():
+            with open('pure.py', 'w') as f, open('crap', 'x'):
+                f.write(new_data)
 
-        sdist_path = mesonpy.build_sdist(os.fspath(tmpdir))
+            sdist_path = mesonpy.build_sdist(os.fspath(tmpdir))
     finally:
         with open('pure.py', 'w') as f:
             f.write(old_data)

